pulpcore/cli/core/config.py

```diff
@@ -0,0 +1,100 @@
+from pathlib import Path
+from typing import Optional
+
+import click
+import toml
+
+LOCATION = str(Path(click.utils.get_app_dir("pulp"), "settings.toml"))
+SETTINGS = ["base_url", "username", "password", "verify_ssl", "format"]
+
+
+@click.group(name="config", help="Manage pulp-cli config file")
+def config() -> None:
+    pass
+
+
+@config.command(help="Create a pulp-cli config settings file")
+@click.option("--interactive", "-i", is_flag=True)
+@click.option("--editor", "-e", is_flag=True, help="Edit the config file in an editor")
+@click.option("--overwrite", "-o", is_flag=True, help="Overwite any existing config file")
+@click.option("--location", default=LOCATION, type=click.Path(resolve_path=True))
+@click.option("--base-url", default="https://localhost", help="API base url")
+@click.option("--username", default="", help="Username on pulp server")
+@click.option("--password", default="", help="Password on pulp server")
+@click.option("--verify-ssl/--no-verify-ssl", default=True, help="Verify SSL connection")
+@click.option(
+    "--format", type=click.Choice(["json", "yaml", "none"], case_sensitive=False), default="json"
+)
+def create(
+    interactive: bool,
+    editor: bool,
+    overwrite: bool,
+    location: str,
+    base_url: str,
+    username: Optional[str],
+    password: Optional[str],
+    verify_ssl: bool,
+    format: str,
+) -> None:
+    if interactive:
+        location = click.prompt("Config location", default=LOCATION)
+        base_url = click.prompt("API base url", default="https://localhost")
+        username = click.prompt("Username", default="")
+        password = click.prompt("Password", default="")
```

Comment:
We could fetch the defaults from the context below, i think. It would guarantee some consistency.

Reply:
Which context?

Reply:
I was thinking about this `ctx.default_map`.

https://github.com/pulp/pulp-cli/blob/develop/pulpcore/cli/common/__init__.py#L23